File: progra3-0583-equipo5-frontend/Frontend_C#/SoftPac/SoftPacModel/AcreedoresDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Remoting.Metadata.W3cXsd2001;
using System.Text;
using System.Threading.Tasks;

namespace SoftPac.Model
{
    public class AcreedoresDTO : EliminableDTOBase
    {
        private int? acreedorId;
        private String razonSocial;
        private String ruc;
        private String direccionFiscal;
        private String condicion;
        private int? plazoDePago;
        private Boolean activo;
        private PaisesDTO pais;

        public AcreedoresDTO() : base()
        {
            this.Pais = new PaisesDTO();
            AcreedorId = null;
            RazonSocial = null;
            Ruc = null;
            DireccionFiscal = null;
            Condicion = null;
            PlazoDePago = null;
            Activo = false;
        }

        public AcreedoresDTO(int acreedor_id, string razon_social, string ruc, string direccion_fiscal, string condicion, int plazo_de_pago, bool activo, PaisesDTO pais) : base()
        {
            this.AcreedorId = acreedor_id;
            this.RazonSocial = razon_social;
            this.Ruc = ruc;
            this.DireccionFiscal = direccion_fiscal;
            this.Condicion = condicion;
            this.PlazoDePago = plazo_de_pago;
            this.Activo = activo;
            this.Pais = pais;
        }

        public AcreedoresDTO(AcreedoresDTO other) : base(other)
        {
            this.AcreedorId = other.AcreedorId;
            this.RazonSocial = other.RazonSocial;
            this.Ruc = other.Ruc;
            this.DireccionFiscal = other.DireccionFiscal;
            this.Condicion = other.Condicion;
            this.PlazoDePago = other.PlazoDePago;
            this.Activo = other.Activo;
            this.Pais = new PaisesDTO(other.Pais);
        }

        public int? AcreedorId { get => acreedorId; set => acreedorId = value; }
        public string RazonSocial { get => razonSocial; set => razonSocial = value; }
        public string Ruc { get => ruc; set => ruc = value; }
        public string DireccionFiscal { get => direccionFiscal; set => direccionFiscal = value; }
        public string Condicion { get => condicion; set => condicion = value; }
        public int? PlazoDePago { get => plazoDePago; set => plazoDePago = value; }
        public bool Activo { get => activo; set => activo = value; }
        public PaisesDTO Pais { get => pais; set => pais = value; }
    }
}

File: BackEnd/Backend_Java/SoftPacModel/src/main/java/pe/edu/pucp/softpac/model/AcreedoresDTO.java
package pe.edu.pucp.softpac.model;


public class AcreedoresDTO extends EliminableDTOBase {
    private Integer acreedor_id;
    private String razon_social;
    private String ruc;
    private String direccion_fiscal;
    private String condicion;
    private Integer plazo_de_pago;
    private Boolean activo;
    private PaisesDTO pais;

    public AcreedoresDTO() {
        super();
        acreedor_id=null;
        razon_social=null;
        ruc=null;
        direccion_fiscal=null;
        pais=null;
        condicion=null;
        plazo_de_pago=0;
    }

    public AcreedoresDTO(Integer acreedor_id, String razon_social, String ruc, String direccion_fiscal, PaisesDTO pais,
            String condicion, Integer plazo_de_pago, Boolean activo) {
        super();
        this.acreedor_id = acreedor_id;
        this.razon_social = razon_social;
        this.ruc = ruc;
        this.direccion_fiscal = direccion_fiscal;
        this.pais = new PaisesDTO(pais);
        this.condicion = condicion;
        this.plazo_de_pago = plazo_de_pago;
        this.activo = activo;
    }
    
    public AcreedoresDTO(AcreedoresDTO acreedor){
        super(acreedor);
        this.acreedor_id = acreedor.getAcreedor_id();
        this.razon_social = acreedor.getRazon_social();
        this.ruc = acreedor.getRuc();
        this.direccion_fiscal = acreedor.getDireccion_fiscal();
        this.pais = acreedor.getPais();
        this.condicion = acreedor.getCondicion();
        this.plazo_de_pago = acreedor.getPlazo_de_pago();
        this.activo = acreedor.getActivo();
    }

    /**
     * @return the acreedor_id
     */
    public Integer getAcreedor_id() {
        return acreedor_id;
    }

    /**
     * @param acreedor_id the acreedor_id to set
     */
    public void setAcreedor_id(Integer acreedor_id) {
        this.acreedor_id = acreedor_id;
    }

    /**
     * @return the razon_social
     */
    public String getRazon_social() {
        return razon_social;
    }

    /**
     * @param razon_social the razon_social to set
     */
    public void setRazon_social(String razon_social) {
        this.razon_social = razon_social;
    }

    /**
     * @return the ruc
     */
    public String getRuc() {
        return ruc;
    }

    /**
     * @param ruc the ruc to set
     */
    public void setRuc(String ruc) {
        this.ruc = ruc;
    }

    /**
     * @return the direccion_fiscal
     */
    public String getDireccion_fiscal() {
        return direccion_fiscal;
    }

    /**
     * @param direccion_fiscal the direccion_fiscal to set
     */
    public void setDireccion_fiscal(String direccion_fiscal) {
        this.direccion_fiscal = direccion_fiscal;
    }

    /**
     * @return the pais
     */
    public PaisesDTO getPais() {
        return pais;
    }

    /**
     * @param pais the pais to set
     */
    public void setPais(PaisesDTO pais) {
        this.pais = pais;
    }

    /**
     * @return the condicion
     */
    public String getCondicion() {
        return condicion;
    }

    /**
     * @param condicion the condicion to set
     */
    public void setCondicion(String condicion) {
        this.condicion = condicion;
    }

    /**
     * @return the plazo_de_pago
     */
    public Integer getPlazo_de_pago() {
        return plazo_de_pago;
    }

    /**
     * @param plazo_de_pago the plazo_de_pago to set
     */
    public void setPlazo_de_pago(Integer plazo_de_pago) {
        this.plazo_de_pago = plazo_de_pago;
    }

    /**
     * @return the activo
     */
    public Boolean getActivo() {
        return activo;
    }

    /**
     * @param activo the activo to set
     */
    public void setActivo(Boolean activo) {
        this.activo = activo;
    }
}
